core-web/src/main/java/org/silverpeas/core/webapi/search/SearchResource.java

```diff
@@ -0,0 +1,105 @@
+package org.silverpeas.core.webapi.search;
+
+import org.silverpeas.core.annotation.RequestScoped;
+import org.silverpeas.core.annotation.Service;
+import org.silverpeas.core.index.search.model.QueryDescription;
+import org.silverpeas.core.index.search.model.SearchResult;
+import org.silverpeas.core.search.SearchService;
+import org.silverpeas.core.util.StringUtil;
+import org.silverpeas.core.webapi.base.RESTWebService;
+import org.silverpeas.core.webapi.base.annotation.Authenticated;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Nicolas Eysseric
+ */
+@Service
+@RequestScoped
+@Path("search")
+@Authenticated
+public class SearchResource extends RESTWebService {
+
+  static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<ResultEntity> search(@QueryParam("query") String query,
+      @QueryParam("taxonomyPosition") String position,
+      @QueryParam("spaceId") String spaceId, @QueryParam("appId") String appId,
+      @QueryParam("startDate") String startDate, @QueryParam("endDate") String endDate) {
+    QueryDescription queryDescription = new QueryDescription(query);
+    queryDescription.setTaxonomyPosition(position);
+
+    if (StringUtil.isDefined(startDate)) {
+      try {
+        startDate = Instant.ofEpochMilli(Long.valueOf(startDate)).atZone(ZoneId.systemDefault()).toLocalDate().format(formatter);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "startDate". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1226)